### PR TITLE
[kube-prometheus-stack] fix: rename thanos sidecar servicemonitor

### PR DESCRIPTION
**Pull request by aslafy-z**
Signed-off-by: Zadkiel Aharonian <zadkiel_aharonian@carrefour.com>

#### What this PR does / why we need it:

Renames the sidecar thanos ServiceMonitor from `{fullname}-thanos-discovery` to `{fullname}-thanos-sidecar`.
This would make the official Thanos Sidecar grafana dashboard work out of the box with this implementation. See https://github.com/thanos-io/thanos/blob/v0.25.2/examples/dashboards/sidecar.json#L1462. 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 36.2.0
+version: 36.3.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 36.2.1
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (aslafy-z) [reply charts/kube-prometheus-stack/Chart.yaml]: You're right. Done!

**Review by monotek — APPROVED** [review]

**Comment by rouke-broersma** [comment]
@monotek @aslafy-z maybe I'm missing something here but if the servicemonitor is changed but the thanos sidecar service is not, how is this change supposed to work? Seems to me there is now a disconnect between the thanos sidecar service and the servicemonitor since the service name is still discovery instead of sidecar.

**Comment by rouke-broersma** [comment]
> @rouke-broersma The ServiceMonitor name has no need to match the Service name. If the selector is well defined, it should not cause any issues AFAIK

But the service label is still thanos-discovery, it does not match with the new selector thanos-sidecar in the servicemonitor.

**Comment by aslafy-z** [comment]
@rouke-broersma I just opened https://github.com/prometheus-community/helm-charts/pull/2244 to fix the issue. Sorry for the noise!

**Pull request merged**